lib/internal/crypto/x509.js

```diff
@@ -61,6 +65,51 @@ let lazyTranslatePeerCertificate;
 
 const kInternalState = Symbol('kInternalState');
 
+const ASN1_months = [
+  'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
+  'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec',
+];
+
+// Parse the given certificate validity time according to the OpenSSL implementation of `ASN1_TIME_print`.
+// MMM dd hh:mm:ss.ms yyyy GMT
+// It is fine to assume that the timezone is in GMT(Zulu), according to RFC 5280.
+function parseASN1Time(value = '') {
+  if (value.length === 0) {
+    return undefined;
+  }
+
+  const date = new Date();
+
+  // Replace all multiple contiguous spaces with a single space
+  value = value.replace(/ +/g, ' ');
+  const arr = StringPrototypeSplit(value, ' ');
+
+  if (!ArrayPrototypeIncludes(ASN1_months, arr[0])) {
+    return undefined;
+  }
+
+  const mon = ASN1_months.indexOf(arr[0]);
+  const day = NumberParseInt(arr[1]);
+  const year = NumberParseInt(arr[3]);
+  date.setUTCFullYear(year, mon, day);
+
+  const time = StringPrototypeSplit(arr[2], ':');
+  const hours = NumberParseInt(time[0]);
+  const min = NumberParseInt(time[1]);
+  const timeSeconds = StringPrototypeSplit(time[2], '.');
+  const sec = NumberParseInt(timeSeconds[0]);
+  date.setUTCHours(hours, min, sec);
+
+  if (timeSeconds.length > 1) {
+    const ms = NumberParseInt(timeSeconds[1]);
+    date.setUTCMilliseconds(ms);
+  } else {
+    date.setUTCMilliseconds(0);
+  }
```

Comment:
It is probably better to implement this at the C++ level in the internal `src/crypto/x509.cc`.